SteamAuth/AuthenticatorLinker.cs

```diff
@@ -52,12 +53,17 @@ public LinkResult AddAuthenticator()
             if (!hasPhone && PhoneNumber == null)
                 return LinkResult.MustProvidePhoneNumber;
 
-            if (!hasPhone)
-            {
-                if (!_addPhoneNumber())
-                {
+			if (!hasPhone) {
+				if (confirmationEmailSent) {
+					if (!_checkEmailConfirmation()) {
                     return LinkResult.GeneralFailure;
                 }
+				} else if (!_addPhoneNumber()) {
+					return LinkResult.GeneralFailure;
+				} else {
+					confirmationEmailSent = true;
+					return LinkResult.MustConfirmEmail;
+				}
```

Comment:
Not a huge issue, but you are using tabs to indent, which is inconsistent with the rest of the file which uses spaces.

Reply:
Thank you, fixed it.